src/processors/AsyncAPIInputProcessor.ts

```diff
@@ -42,72 +43,81 @@ export class AsyncAPIInputProcessor extends AbstractInputProcessor {
    */
   // eslint-disable-next-line sonarjs/cognitive-complexity
   static reflectSchemaNames(
-    schema: AsyncAPISchema | boolean
+    schema: AsyncAPISchema | boolean,
+    alreadyIteratedSchemas: Map<string, Schema> = new Map()
   ): Schema | boolean {
     if (typeof schema === 'boolean') return schema;
+    const schemaUid = schema.uid();
+    if (alreadyIteratedSchemas.has(schemaUid)) {
+      const cachedOutputSchema = alreadyIteratedSchemas.get(schemaUid); 
+      if (cachedOutputSchema !== undefined) {
+        return cachedOutputSchema;
+      }
+    }
```

Comment:
If you call first `has` and then `get` you have sure that the key exists and also I don't think so that `cachedOutputSchema` can be undefined in some cases.

```suggestion
    if (alreadyIteratedSchemas.has(schemaUid)) {
      return alreadyIteratedSchemas.get(schemaUid); 
    }
```